cmake/FindHerbert.cmake

```diff
@@ -31,39 +34,55 @@ set(DIRS_TO_SEARCH
     ENV{ProgramFiles}/Herbert
 )
 
-# Always call this find_path as it make Herbert_ROOT editable in the CMake GUI
-find_path(Herbert_ROOT
+# If given defined Herbert root as arg.
+if (Herbert_ROOT)
+  set(DIRS_TO_SEARCH ${DIRS_TO_SEARCH} ${Herbert_ROOT})
+endif()
+
+# Always call this find_path as it make Herbert_CORE editable in the CMake GUI
+find_path(Herbert_CORE
     NAMES "herbert_init.m"
     PATHS ${DIRS_TO_SEARCH}
     PATH_SUFFIXES "herbert_core"
-    DOC "The Herbert root directory - the directory containing herbert_init.m"
+    DOC "The Herbert core directory - the directory containing herbert_init.m"
 )
 
-# Deals with the case where Herbert_ROOT is given on the command line but to
+# Deals with the case where Herbert_CORE is given on the command line but to
 # somewhere incorrect
-if(Herbert_ROOT AND NOT EXISTS "${Herbert_ROOT}/herbert_init.m")
+if(Herbert_CORE AND NOT EXISTS "${Herbert_CORE}/herbert_init.m")
     message(FATAL_ERROR
-        "Could not find herbert_init.m inside Herbert_ROOT: '${Herbert_ROOT}'")
+        "Could not find herbert_init.m inside Herbert_CORE: '${Herbert_CORE}'")
+endif()
+
+if (NOT Herbert_ROOT)
+  # Root is one dir above core
+  set(Herbert_ROOT "${Herbert_CORE}/.." CACHE PATH "Directory containing Herbert")
+else()
+  # If given through CLI
+  set(Herbert_ROOT "${Herbert_ROOT}" CACHE PATH "Directory containing Herbert")
 endif()
+mark_as_advanced(Herbert_ROOT)
```

Comment:
why is this advanced?

Reply:
Because it should be set automatically, not generally by the user unless they know what they're doing.